components/popover_list_members/popover_list_members.tsx

```diff
@@ -0,0 +1,293 @@
+            status={props.statuses[user.id]}
+            user={user}
+        />
+    ));
+
+    const channelIsArchived = props.channel.delete_at !== 0;
+    let popoverButton;
+    let handleButtonOnClick = showMembersModal;
+    let editButton = null;
+    if (props.channel.type !== Constants.GM_CHANNEL && !channelIsArchived) {
+        let membersName = (
+            <FormattedMessage
+                id='members_popover.manageMembers'
+                defaultMessage='Manage Members'
+            />
+        );
+
+        const isDefaultChannel = props.channel.name === Constants.DEFAULT_CHANNEL;
+
+        if (isDefaultChannel || !props.manageMembers) {
+            membersName = (
+                <FormattedMessage
+                    id='members_popover.viewMembers'
+                    defaultMessage='View Members'
+                />
+            );
+        }
+
+        if (props.addMembersABTest === AddMembersToChanneltreatments.BOTTOM && props.manageMembers) {
+            handleButtonOnClick = () => onAddNewMembersButton(AddMembersToChanneltreatments.BOTTOM);
+            membersName = (
+                <FormattedMessage
+                    id='members_popover.addMembers'
+                    defaultMessage='Add Members'
+                />
+            );
+            editButton = (
+                <button
+                    className='btn btn-link'
+                    id='editBtn'
+                    onClick={showMembersModal}
+                >
+                    <i className='icon icon-pencil-outline'/>
+                    <FormattedMessage
+                        id='members_popover.editBtn'
+                        defaultMessage='Edit'
+                    />
+                </button>
+            );
+        } else if (props.addMembersABTest === AddMembersToChanneltreatments.TOP && props.manageMembers) {
+            editButton = (
+                <button
+                    className='btn btn-link'
+                    id='addBtn'
+                    onClick={() => onAddNewMembersButton(AddMembersToChanneltreatments.TOP)}
+                >
+                    <i className='icon icon-account-plus-outline'/>
+                    <FormattedMessage
+                        id='members_popover.add'
+                        defaultMessage='Add'
+                    />
+                </button>
+            );
+        }
+
+        popoverButton = (
+            <div
+                className='more-modal__button'
+                key={'popover-member-more'}
+            >
+                <button
+                    className='btn btn-link'
+                    data-testid='membersModal'
+                    onClick={handleButtonOnClick}
+                >
+                    {membersName}
+                </button>
+            </div>
+        );
+    }
+    const count = props.memberCount;
+    let countText = '-';
+    if (count > 0) {
+        countText = count.toString();
+    }
```

Comment:
this is too complicated:

```suggestion
    const countText = props.memberCount > 0 ? props.memberCount.toString() : '-';
```

Reply:
Done